### ISSUE TITLE
compatibility issue (training error/accuracy) with tensorflow 1.5

### ISSUE THREAD

**Issue by jwnsu** [endernewton/tf-faster-rcnn#290]
Everything works fine with tensorflow 1.4. 

For newly released tensorflow 1.5, the training loss became NaN at 80 iterations:

```
iter: 80 / 84000, total loss: nan
 >>> rpn_loss_cls: 0.684417
 >>> rpn_loss_box: 0.019884
 >>> loss_cls: 0.101328
 >>> loss_box: 0.000000
 >>> lr: 0.001000
```

**Comment by tedivm** [comment]
Did you find out what was causing this? We're running into a similar issue.

**Comment by jwnsu** [comment]
It seems tensorflow 1.5 affects training in subtle way. Regarding the error, it goes away after minor adjustment (e.g. learning rate.) However, now we find accuracy trained with tf1.5 is a couple of percentage points lower than tf1.4, still figuring out what's the cause.

**Comment by autoalpha** [comment]
@jwnsu  I'm also debugging with this issue, and it seems like the optimizer computes NaN gradients for some reason. What optimizer are you using?

**Comment by jwnsu** [comment]
@autoalpha MomentumOptimizer

**Comment by autoalpha** [comment]
@jwnsu  I'm using AdamOptimizer. The NaN problem goes away when I use cpu version of tensorflow 1.5. This makes me think it may be a bug of new cuda9 / cudnn7.

**Comment by jwnsu** [comment]
CUDA 9.1 is already available, it may take a while for future tf version to support it though. For the time being, we can stick with tf1.4 for this repository.
